napari/utils/colormaps/colormaps.py

```diff
@@ -111,6 +118,36 @@ def _low_discrepancy_image(image, seed=0.5, margin=1 / 256):
     return image_out
 
 
+def color_dict_to_colormap(color_dict):
+    """
+    Generate a color map based on the given color dictionary
+    Parameters
+    ----------
+    color_dict : dict of int
+        Mapping between labels and color strings
+
+    Returns
+    -------
+    colormap : Colormap
+        Colormap constructed with provided control colors
+    label_color_index : dict of int
+        Mapping of Label to color control point within colormap
+    """
+    colors = [
+        transform_color(color_str)[0]
```

Comment:
See comment about about moving `transform_color(color_str)[0]` into the dict setter